x-pack/plugins/observability_solution/apm/public/components/alerting/ui_components/apm_rule_unified_search_bar.tsx

```diff
@@ -38,7 +38,12 @@ export function ApmRuleUnifiedSearchBar({
 
   const handleSubmit = (payload: { query?: Query }) => {
     const { query } = payload;
-    setRuleParams('searchConfiguration', { query });
+    try {
+      fromKueryExpression(query?.query as string);
+      setRuleParams('searchConfiguration', { query });
+    } catch (e) {
+      // ignore invalid queries
```

Comment:
Shouldn't we show a warning or something in the UI if the query is invalid? The user might not notice it.

Reply:
Good idea. Implemented 
<img width="503" alt="Screenshot 2024-12-05 at 14 44 23" src="https://github.com/user-attachments/assets/28c7e062-606f-44e8-ad0f-e75a91003f3b">